core/device_impl.cpp

```diff
@@ -155,6 +159,50 @@ void DeviceImpl::process_autopilot_version(const mavlink_message_t &message)
     unregister_timeout_handler(_autopilot_version_timed_out_cookie);
 }
 
+void DeviceImpl::process_statustext(const mavlink_message_t &message)
+{
+    mavlink_statustext_t statustext;
+    mavlink_msg_statustext_decode(&message, &statustext);
+
+    std::string debug_str = "mavlink ";
+
+    switch (statustext.severity) {
+        case MAV_SEVERITY_EMERGENCY:
+            debug_str += "emergency";
+            break;
+        case MAV_SEVERITY_ALERT:
+            debug_str += "alert";
+            break;
+        case MAV_SEVERITY_CRITICAL:
+            debug_str += "critical";
+            break;
+        case MAV_SEVERITY_ERROR:
+            debug_str += "error";
+            break;
+        case MAV_SEVERITY_WARNING:
+            debug_str += "warning";
+            break;
+        case MAV_SEVERITY_NOTICE:
+            debug_str += "notice";
+            break;
+        case MAV_SEVERITY_INFO:
+            debug_str += "info";
+            break;
+        case MAV_SEVERITY_DEBUG:
+            debug_str += "debug";
+            break;
+        default:
+            break;
+    }
+
+    // statustext.text is not null terminated, therefore we copy it first to
+    // an array big enough that is zeroed.
+    char text_with_null[sizeof(statustext.text) + 1] {};
+    memcpy(text_with_null, statustext.text, sizeof(statustext.text));
```

Comment:
dirty indeed, but I don't see any other way around this if we can't control the input, and now that I think about it, can this lead to memory leaks? can someone use this to DOS an app using DroneCore by crafting big malicious messages?

Reply:
I think all that a malicious drone can do here is to print a bunch of memory that follows after `statustext.text` until `0` appears.

Reply:
In conclusion, yes we make ourselves open to a crafty mavlink attack, but I'm sure this isn't the only place where this occurs